scripts/build_desktop_app_with_firebase.py

```diff
@@ -0,0 +1,281 @@
+    # Its a good habit to specify for x64 too as the default might be different
+    # on different windows machines.
+    cmd.append('-A')
+    cmd.append('Win32') if arch == 'x86' else cmd.append('x64')
+
+    # Use our special cmake option for /MD (dynamic).
+    # If this option is not specified, the default value is /MT (static).
+    if msvc_runtime_library == "static":
+      cmd.append('-DMSVC_RUNTIME_LIBRARY_STATIC=ON')
+
+  if (target_format):
+    cmd.append('-DFIREBASE_XCODE_TARGET_FORMAT={0}'.format(target_format))
+  print("Running {0}".format(' '.join(cmd)))
+  subprocess.run(cmd, cwd=app_dir, check=True)
+
+  #CMake build
+  cmd = ['cmake', '--build', build_dir, '-j', str(os.cpu_count()), '--config', config]
+  print("Running {0}".format(' '.join(cmd)))
+  subprocess.run(cmd, cwd=app_dir, check=True)
+
+def build_app_with_prebuilt(app_dir, sdk_prebuilt_dir, build_dir, arch,
+                            msvc_runtime_library='static', config=None):
+  """Build desktop app directly against the prebuilt firebase C++ libraries.
+
+  Since this invovles a cmake configure, it is advised to run this on a clean
+  build directory.
+
+  Args:
+   app_dir (str): Path to directory containing application's CMakeLists.txt.
+   sdk_prebuilt_dir (str): Path to prebuilt firebase C++ libraries.
+   build_dir (str): Output build directory.
+   arch (str): Platform Architecture (example: 'x64').
+   msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+   config (str): Release/Debug config.
+          If its not specified, cmake's default is used (most likely Debug).
+  """
+
+  cmd = ['cmake', '-S', '.', '-B', build_dir]
+  cmd.append('-DFIREBASE_CPP_SDK_DIR={0}'.format(sdk_prebuilt_dir))
+
+  if platform.system() == 'Windows':
+    if arch == 'x64':
+      cmd.append('-DCMAKE_CL_64=ON')
+    if msvc_runtime_library == 'dynamic':
+      cmd.append('-DMSVC_RUNTIME_MODE=MD')
+    else:
+      cmd.append('-DMSVC_RUNTIME_MODE=MT')
+
+  if config:
+    cmd.append('-DCMAKE_BUILD_TYPE={0}'.format(config))
+
+  print("Running {0}".format(' '.join(cmd)))
+  subprocess.run(cmd, cwd=app_dir, check=True)
+
+  #CMake build
+  cmd = ['cmake', '--build', build_dir, '-j', str(os.cpu_count()), '--config', config]
+  print("Running {0}".format(' '.join(cmd)))
+  subprocess.run(cmd, cwd=app_dir, check=True)
+
+def main():
+  args = parse_cmdline_args()
+
+  if not is_path_valid_for_cmake(args.sdk_dir):
+    print ("SDK path provided is not valid. Could not find a CMakeLists.txt at the root level.\n"
+           "Please check the argument to '--sdk_dir'")
+    sys.exit(1)
+
+  if not is_path_valid_for_cmake(args.app_dir):
+    print ("App path provided is not valid. Could not find a CMakeLists.txt at the root level.\n"
+           "Please check the argument to '--app_dir'")
+    sys.exit(1)
+
+  if is_sdk_path_source(args.sdk_dir):
+    print("SDK path provided is Firebase C++ source directory. Building...")
```

Comment:
Nit: "is a Firebase C++ source directory"

Reply:
Done.